### PR TITLE
return exactly err on cursor decode results

### PR DESCRIPTION
**Pull request by suhli**
ref: [return err when cursor decodes](https://github.com/qiniu/qmgo/issues/287)

### CODE REVIEW

File: cursor.go [hunk 1]
```diff
@@ -28,18 +28,19 @@ type Cursor struct {
 
 // Next gets the next document for this cursor. It returns true if there were no errors and the cursor has not been
```
Comment (jiangz222) [inline cursor.go:29]: Please fix comments

File: cursor.go [hunk 2]
```diff
@@ -28,18 +28,19 @@ type Cursor struct {
...
 // exhausted.
-func (c *Cursor) Next(result interface{}) bool {
+func (c *Cursor) Next(result interface{}) error {
 	if c.err != nil {
-		return false
+		return c.err
 	}
 	var err error
 	if c.cursor.Next(c.ctx) {
 		err = c.cursor.Decode(result)
```
Comment (jiangz222) [inline cursor.go:37]: Just  go with `return c.cursor.Decode(result)`

**Comment by jiangz222** [comment]
Hi @suhli, I checked the implement of `cursor.Next(...)` in mongo-go-driver, the error is return by 
```
func (c *Cursor) Err() error { return c.err }
```
And qmgo has function `Err()` too, so I think you should pass error by `Err()` with `cursor.err`, codes may like
```
func (c *Cursor) Next(result interface{}) bool {
......
	if c.cursor.Next(c.ctx) {
		err = c.cursor.Decode(result)
		if err != nil {
			c.err = err
			return false
		}
		return true
	}
......
}
```
Please share your idea.

**Comment by suhli** [comment]
> 



> Hi @suhli, I checked the implement of `cursor.Next(...)` in mongo-go-driver, the error is return by
> 
> ```
> func (c *Cursor) Err() error { return c.err }
> ```
> 
> And qmgo has function `Err()` too, so I think you should pass error by `Err()` with `cursor.err`, this pr should change codes like:
> 
> ```
> func (c *Cursor) Next(result interface{}) bool {
> ......
> 	if c.cursor.Next(c.ctx) {
> 		err = c.cursor.Decode(result)
> 		if err != nil {
> 			c.err = err
> 			return false
> 		}
> 		return true
> 	}
> ......
> }
> ```
> 
> Please share your idea.

sure,but how can we get the decode err outside?
i check cursor. Decode in mongo-go-driver,they don't set err at cursor but just return that
or we can set the err into qmgo.cursor?
so we can do like this outside
```go
...
if !cr.Next(result) {
   if cr.err != nil{
     return cr.err
   }
   return ErrNoSuchDocuments
}
....
```

**Comment by suhli** [comment]
@jiangz222 ok,please check the file changes

**Pull request merged**